src/control/positionControl/pathPlanning/VoronoiPathPlanning.cpp

```diff
@@ -0,0 +1,153 @@
+        return;
+    }
+
+    double minOriginDist = -1;
+    rtt::Vector2 closestOriginPoint;
+    double minTargetDist = -1;
+    rtt::Vector2 closestTargetPoint;
+    //add origin and target to the graph
+    for(auto const& positionIterator : graphAdjacencyList){
+        if(minOriginDist < 0 || (positionIterator.first-robotPosition).length() < minOriginDist){
+            closestOriginPoint = positionIterator.first;
+            minOriginDist = (positionIterator.first-robotPosition).length();
+        }
+        if(minTargetDist < 0 || (positionIterator.first-targetPosition).length() < minTargetDist){
+            closestTargetPoint = positionIterator.first;
+            minTargetDist = (positionIterator.first-targetPosition).length();
+        }
+    }
+    if (graphAdjacencyList.find(robotPosition) == graphAdjacencyList.end()) {
+        graphAdjacencyList.insert(
+                {robotPosition, std::list<GraphNode>(1, (GraphNode) {closestOriginPoint, minOriginDist})});
+    }
+    if (graphAdjacencyList.find(targetPosition) == graphAdjacencyList.end()) {
+        graphAdjacencyList[closestTargetPoint].push_back({targetPosition, minTargetDist});
+    }
+}
+
+const std::unordered_map<rtt::Vector2, std::list<GraphNode>, hashPoint> &
+VoronoiPathPlanning::getGraphAdjacencyList() const {
+    return graphAdjacencyList;
+}
+
+std::list<rtt::Vector2> VoronoiPathPlanning::generatePathDijkstra(const rtt::Vector2& initialPosition, const rtt::Vector2& targetPosition) {
+    std::unordered_map<rtt::Vector2, float, hashPoint> distanceVector;
+    std::unordered_map<rtt::Vector2, rtt::Vector2, hashPoint> parentVector;
+    distanceVector.insert({initialPosition,0});
+    std::list<rtt::Vector2> nodeQueue;
+    nodeQueue.push_front(initialPosition);
+    for(const rtt::Vector2& currentNode: nodeQueue){
+        if (currentNode == targetPosition){
+            break;
+        }
+        for (const GraphNode& adjacentNode : graphAdjacencyList[currentNode]){
+            if (distanceVector.find(adjacentNode.nextNodePosition) == distanceVector.end()) {
+                if (std::find(nodeQueue.begin(), nodeQueue.end(), adjacentNode.nextNodePosition) == nodeQueue.end()) {
+                    nodeQueue.push_back(adjacentNode.nextNodePosition);
+                }
+                distanceVector.insert({adjacentNode.nextNodePosition, distanceVector[currentNode] + adjacentNode.distance});
+                parentVector.insert({adjacentNode.nextNodePosition, currentNode});
+                continue;
+            }
+            if(distanceVector[adjacentNode.nextNodePosition] > distanceVector[currentNode] + adjacentNode.distance){
+                distanceVector[adjacentNode.nextNodePosition] = distanceVector[currentNode] + adjacentNode.distance;
+                parentVector[adjacentNode.nextNodePosition] = currentNode;
+            }
+        }
+    }
```

Comment:
This needs a comment. It's not 'super' complicated but it's certainly not easy either.